core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/AddOperationsToChain.java

```diff
@@ -0,0 +1,227 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.graph.hook;
+
+import uk.gov.gchq.gaffer.commonutil.StreamUtil;
+import uk.gov.gchq.gaffer.data.elementdefinition.exception.SchemaException;
+import uk.gov.gchq.gaffer.exception.SerialisationException;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.store.operationdeclaration.OperationDeclarations;
+import uk.gov.gchq.gaffer.user.User;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A <code>AddOperationsToChain</code> is a {@link GraphHook} that allows a
+ * user to insert additional operations at certain points on the operation chain.
+ * At the start, before a specific Operation, after a specific Operation, or at the end.
+ * A user can also specify authorised Operations to add, and if the user has
+ * the opAuths, the additional Operations will be added to the chain.
+ */
+public class AddOperationsToChain implements GraphHook {
+    private static final JSONSerialiser JSON_SERIALISER = new JSONSerialiser();
+
+    private final AdditionalOperations defaultOperations = new AdditionalOperations();
+    private LinkedHashMap<String, AdditionalOperations> authorisedOps;
+
+    /**
+     * Default constructor.
+     */
+    public AddOperationsToChain() {
+    }
+
+    /**
+     * Constructs a {@link AddOperationsToChain} using the path to json
+     * file specifying Operations to add.
+     *
+     * @param addOperationsPath Path to file containing Operations to add.
+     * @throws IOException if the file reading fails.
+     */
+    public AddOperationsToChain(final String addOperationsPath) throws IOException {
+
+        Path path = Paths.get(addOperationsPath);
+        AddOperationsToChain addOperations;
+        if (path.toFile().exists()) {
+            addOperations = fromJson(Files.readAllBytes(path));
+        } else {
+            addOperations = fromJson(StreamUtil.openStream(OperationDeclarations.class, addOperationsPath));
+        }
+        setOperations(addOperations);
+    }
+
+    /**
+     * Constructs a {@link AddOperationsToChain} using the json byte[]
+     * specifying Operations to add.
+     *
+     * @param addOperationsBytes byte[] containing Operations to add.
+     * @throws IOException if the byte[] fails to be deserialised.
+     */
+    public AddOperationsToChain(final byte[] addOperationsBytes) throws IOException {
+        setOperations(fromJson(addOperationsBytes));
+    }
+
+    public void setStart(final List<Operation> start) {
```

Comment:
Please can you put the getters/setters at the bottom of the public methods. The important methods should go first, i.e. preExecute and postExecute